syft/pkg/cataloger/binary/classifier.go

```diff
@@ -195,3 +262,52 @@ func singleCPE(cpeString string) []cpe.CPE {
 		cpe.Must(cpeString),
 	}
 }
+
+// sharedLibraries returns a list of all shared libraries found within a binary, currently
+// supporting: elf, macho, and windows pe
+func sharedLibraries(resolver source.FileResolver, location source.Location) ([]string, error) {
```

Comment:
I can't think of a better way to write this in terms of "detect the binary format then use the full parser" other than peaking at the magic number, which I think would be overkill here anyway.

Though, since there are a few formats and they share the same `ImportedLibraries ` there could be an ad-hoc interface and a list of constructors of the interface to make this more of a list of the same operation being done instead of it being unraveled, but I don't think there'd be any huge benefit and would probably be the same LoC anyway. 

(this was more thinking "outloud", consider this whole comment non-blocking)

Reply:
Agree, I don't think there would be a huge benefit to changing this at the moment, at least.